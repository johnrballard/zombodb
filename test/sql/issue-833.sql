CREATE TABLE issue833 (
                                    pk bigint NOT NULL,
                                    pizza json
);

CREATE INDEX idxissue833 ON issue833
    USING zombodb ((issue833.*));

INSERT INTO issue833 values (1, '[{"topping":"pepperoni"},{"tastiness":"great"}]');
INSERT INTO issue833 values (2, '[{"topping":"mushroom"},{"tastiness":"good"}]');
INSERT INTO issue833 values (3, '[{"topping":"mozzarella"},{"tastiness":"necessary"}]');
INSERT INTO issue833 values (4, '[{"topping":"pineapple"},{"tastiness":"n/a"}]');

SELECT * FROM zdb.tally('issue833','pizza.topping',True,'^.*','') limit 10;
SELECT * FROM zdb.tally('issue833','pizza.topping',True,'^p.*','') limit 10;
SELECT * FROM zdb.tally('issue833','pizza.topping',True,'^pe.*','pizza.topping = "p*"') limit 10;

DROP TABLE issue833 CASCADE;