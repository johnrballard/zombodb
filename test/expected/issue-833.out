CREATE TABLE issue833 (
                                    pk bigint NOT NULL,
                                    pizza json
);
CREATE INDEX idxissue833 ON issue833
    USING zombodb ((issue833.*));
INSERT INTO issue833 values (1, '[{"topping":"pepperoni"},{"tastiness":"great"}]');
INSERT INTO issue833 values (2, '[{"topping":"mushroom"},{"tastiness":"good"}]');
INSERT INTO issue833 values (3, '[{"topping":"mozzarella"},{"tastiness":"necessary"}]');
INSERT INTO issue833 values (4, '[{"topping":"pineapple"},{"tastiness":"n/a"}]');
SELECT * FROM zdb.tally('issue833','pizza.topping',True,'^.*','') limit 10;
    term    | count 
------------+-------
            |     4
 mozzarella |     1
 mushroom   |     1
 pepperoni  |     1
 pineapple  |     1
(5 rows)

SELECT * FROM zdb.tally('issue833','pizza.topping',True,'^p.*','') limit 10;
   term    | count 
-----------+-------
           |     4
 pepperoni |     1
 pineapple |     1
(3 rows)

SELECT * FROM zdb.tally('issue833','pizza.topping',True,'^pe.*','pizza.topping = "p*"') limit 10;
   term    | count 
-----------+-------
 pepperoni |     1
(1 row)

DROP TABLE issue833 CASCADE;
